sw/otbn/crypto/tests/generated/sha3_testgen.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import random
+from typing import TextIO, Optional
+
+from Crypto.Hash import SHA3_224, SHA3_256, SHA3_384, SHA3_512
+
+from shared.testgen import write_test_data, write_test_exp
+
+DST_LEN = 64 # 28, 32, 48, 64
```

Comment:
Should this be a random choice rather than a fixed value, so that variants other than SHA3-512 get tested?

Reply:
I initially thought that all 10 tests are run for each variant of SHA-3 to test its correctness, and thus the choice for the variant can be set manually. If I understood correctly, in case `DST_LEN` is set to be random, maybe only 2 tests for SHA3-512 are ran, while 4 tests for SHA3-224 are run, and maybe 0 test for SHA3-256, etc. Is this enough for each variant? Which test scenario that you want to have?

Reply:
Every time a PR is submitted to OpenTitan, the test will run via Bazel automatically as part of CI, and the test vectors will be generated by this script (i.e. only for SHA3-512). It would be nice if these automatic tests captured multiple variants, since they will be run much more often than anyone actually runs the tests manually. For that reason, I'd rather do it automatically instead of via arguments. I'm okay with either selecting it randomly or simply iterating through the variants as it generates the 10 tests. It's not strictly necessary to cover every variant every time; the manual `.exp` test already checks all the variants in turn, so in this case I think the random test is just an extra layer of checks to catch things the manual tests may have missed. :slightly_smiling_face: